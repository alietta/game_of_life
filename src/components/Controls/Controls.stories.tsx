import React from 'react';
import { action } from '@storybook/addon-actions';
import { DarkMode, LightMode, Layer } from 'sancho';
import { Controls } from './index';

export default { title: 'controls' };

export const GameControls = () => (
  <div css={{ width: 400 }}>
    <DarkMode>
      <Layer css={{ borderRadius: 0 }}>
        <Controls resetGame={action('reset')} changeSpeed={action('changeSpeed')} />
      </Layer>
    </DarkMode>
    <LightMode>
      <Layer css={{ borderRadius: 0 }}>
        <Controls resetGame={action('reset')} changeSpeed={action('changeSpeed')} />
      </Layer>
    </LightMode>
  </div>
);
