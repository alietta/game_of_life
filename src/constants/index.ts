export const PLAY = 'play';
export const SLOW = 'slow';
export const FAST = 'fast';
export const PAUSE = 'pause';
export const RESET = 'reset';
