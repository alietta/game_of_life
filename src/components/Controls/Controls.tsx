import React, { FC } from 'react';
import { PLAY, FAST, SLOW, PAUSE } from '@/constants';
import {
  Layer,
  IconPause,
  IconFastForward,
  IconPlay,
  IconRewind,
  IconRefreshCw,
  Button,
} from 'sancho';

export interface ControlsProps {
  resetGame: () => void;
  changeSpeed: (speed: string) => void;
}
const Controls: FC<ControlsProps> = props => {
  const { resetGame, changeSpeed } = props;
  const handleClick = (speed: string) => {
    return () => {
      changeSpeed(speed);
    };
  };

  return (
    <Layer elevation="xs">
      <Button
        size="lg"
        name="reset"
        intent="primary"
        variant="outline"
        onClick={resetGame}
      >
        <IconRefreshCw />
      </Button>
      <Button
        size="lg"
        name={PAUSE}
        intent="primary"
        variant="outline"
        onClick={handleClick(PAUSE)}
      >
        <IconPause />
      </Button>
      <Button
        size="lg"
        name={PLAY}
        intent="primary"
        variant="outline"
        onClick={handleClick(PLAY)}
      >
        <IconPlay />
      </Button>
      <Button
        size="lg"
        name={SLOW}
        intent="primary"
        variant="outline"
        onClick={handleClick(SLOW)}
      >
        <IconRewind />
      </Button>
      <Button
        size="lg"
        name={FAST}
        intent="primary"
        variant="outline"
        onClick={handleClick(FAST)}
      >
        <IconFastForward />
      </Button>
    </Layer>
  );
};
export { Controls };
