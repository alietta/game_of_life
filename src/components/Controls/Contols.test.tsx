import React from 'react';
import { mount } from 'enzyme';
import { RESET, PLAY, FAST, SLOW, PAUSE } from '@/constants';
import { Controls } from './Controls';

describe('Controls', () => {
  it('test render', () => {
    const resetGame = jest.fn();
    const changeSpeed = jest.fn();
    const element = mount(<Controls resetGame={resetGame} changeSpeed={changeSpeed} />);
    expect(element).toMatchSnapshot();
  });
  it('should call reset with props', () => {
    const resetGame = jest.fn();
    const changeSpeed = jest.fn();
    const element = mount(
      <Controls resetGame={resetGame} changeSpeed={changeSpeed} />
    );
    element.find(`div[name='${RESET}']`).simulate('click');
    expect(resetGame).toHaveBeenCalled();
  });
  it('should call pause with props', () => {
    const resetGame = jest.fn();
    const changeSpeed = jest.fn();
    const element = mount(
      <Controls resetGame={resetGame} changeSpeed={changeSpeed} />
    );
    element.find(`div[name='${PAUSE}']`).simulate('click');
    expect(changeSpeed).toHaveBeenCalledWith(PAUSE);
  });
  it('should call play with props', () => {
    const resetGame = jest.fn();
    const changeSpeed = jest.fn();
    const element = mount(
      <Controls resetGame={resetGame} changeSpeed={changeSpeed} />
    );
    element.find(`div[name='${PLAY}']`).simulate('click');
    expect(changeSpeed).toHaveBeenCalledWith(PLAY);
  });
  it('should call fast speed with props', () => {
    const resetGame = jest.fn();
    const changeSpeed = jest.fn();
    const element = mount(
      <Controls resetGame={resetGame} changeSpeed={changeSpeed} />
    );
    element.find(`div[name='${FAST}']`).simulate('click');
    expect(changeSpeed).toHaveBeenCalledWith(FAST);
  });
  it('should call slow speed with props', () => {
    const resetGame = jest.fn();
    const changeSpeed = jest.fn();
    const element = mount(
      <Controls resetGame={resetGame} changeSpeed={changeSpeed} />
    );
    element.find(`div[name='${SLOW}']`).simulate('click');
    expect(changeSpeed).toHaveBeenCalledWith(SLOW);
  });
});
