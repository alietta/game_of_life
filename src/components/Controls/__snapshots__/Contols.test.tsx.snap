// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameControl test render 1`] = `
.emotion-6 {
  border-radius: 0;
  box-shadow: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-6 .Button {
  margin: 0 5px;
}

.emotion-6 .Button:first-of-type {
  margin-left: 0px;
}

.emotion-6 .Button:last-of-type {
  margin-right: 0px;
}

.emotion-5 {
  position: relative;
  background: white;
  box-shadow: 0 0 1px hsla(210,10.3%,22.7%,0.1), 0 0 1px 1px hsla(210,10.3%,22.7%,0.12);
  border-radius: 1rem;
  border-radius: 0;
  box-shadow: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-5 .Button {
  margin: 0 5px;
}

.emotion-5 .Button:first-of-type {
  margin-left: 0px;
}

.emotion-5 .Button:last-of-type {
  margin-right: 0px;
}

.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  background: none;
  white-space: nowrap;
  -webkit-appearance: none;
  box-sizing: border-box;
  text-align: center;
  -webkit-font-smoothing: antialiased;
  border: none;
  text-rendering: optimizeLegibility;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: transparent;
  cursor: pointer;
  box-shadow: none;
  font-weight: 500;
  position: relative;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";
  border-radius: 0.25rem;
  font-size: 1.25rem;
  padding: 0 1.5rem;
  height: 48px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 1px solid;
  background: none;
  color: #1971c2;
  border-color: hsla(208.79999999999995,77.2%,42.9%,0.2);
  box-shadow: none;
  -webkit-transition: box-shadow 0.07s cubic-bezier(0.35,0,0.25,1),background 0.07s cubic-bezier(0.35,0,0.25,1);
  transition: box-shadow 0.07s cubic-bezier(0.35,0,0.25,1),background 0.07s cubic-bezier(0.35,0,0.25,1);
}

.emotion-0:focus {
  outline: 3px auto hsla(208.79999999999995,77.2%,42.9%,0.8);
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
}

.emotion-0[aria-expanded="true"] {
  background: hsla(208.79999999999995,77.2%,42.9%,0.15);
}

<Controls
  changeSpeed={[MockFunction]}
  resetGame={[MockFunction]}
>
  <Layer
    className="emotion-6"
    elevation="xs"
  >
    <div
      className="emotion-5"
    >
      <Button
        intent="primary"
        name="reset"
        onClick={[MockFunction]}
        size="lg"
        variant="outline"
      >
        <div
          className="Button Touchable emotion-0"
          name="reset"
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseDownCapture={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchMoveCapture={[Function]}
          onTouchStart={[Function]}
          onTouchStartCapture={[Function]}
          role="button"
          tabIndex={0}
        >
          <Component>
            <svg
              fill="none"
              height="20px"
              stroke="#212529"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="20px"
            >
              <polyline
                points="23 4 23 10 17 10"
              />
              <polyline
                points="1 20 1 14 7 14"
              />
              <path
                d="M3.51 9a9 9 0 0 1 14.85-3.36L23 10M1 14l4.64 4.36A9 9 0 0 0 20.49 15"
              />
            </svg>
          </Component>
        </div>
      </Button>
      <Button
        intent="primary"
        name="pause"
        onClick={[Function]}
        size="lg"
        variant="outline"
      >
        <div
          className="Button Touchable emotion-0"
          name="pause"
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseDownCapture={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchMoveCapture={[Function]}
          onTouchStart={[Function]}
          onTouchStartCapture={[Function]}
          role="button"
          tabIndex={0}
        >
          <Component>
            <svg
              fill="none"
              height="20px"
              stroke="#212529"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="20px"
            >
              <rect
                height="16"
                width="4"
                x="6"
                y="4"
              />
              <rect
                height="16"
                width="4"
                x="14"
                y="4"
              />
            </svg>
          </Component>
        </div>
      </Button>
      <Button
        intent="primary"
        name="play"
        onClick={[Function]}
        size="lg"
        variant="outline"
      >
        <div
          className="Button Touchable emotion-0"
          name="play"
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseDownCapture={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchMoveCapture={[Function]}
          onTouchStart={[Function]}
          onTouchStartCapture={[Function]}
          role="button"
          tabIndex={0}
        >
          <Component>
            <svg
              fill="none"
              height="20px"
              stroke="#212529"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="20px"
            >
              <polygon
                points="5 3 19 12 5 21 5 3"
              />
            </svg>
          </Component>
        </div>
      </Button>
      <Button
        intent="primary"
        name="slow"
        onClick={[Function]}
        size="lg"
        variant="outline"
      >
        <div
          className="Button Touchable emotion-0"
          name="slow"
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseDownCapture={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchMoveCapture={[Function]}
          onTouchStart={[Function]}
          onTouchStartCapture={[Function]}
          role="button"
          tabIndex={0}
        >
          <Component>
            <svg
              fill="none"
              height="20px"
              stroke="#212529"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="20px"
            >
              <polygon
                points="11 19 2 12 11 5 11 19"
              />
              <polygon
                points="22 19 13 12 22 5 22 19"
              />
            </svg>
          </Component>
        </div>
      </Button>
      <Button
        intent="primary"
        name="fast"
        onClick={[Function]}
        size="lg"
        variant="outline"
      >
        <div
          className="Button Touchable emotion-0"
          name="fast"
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseDownCapture={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchMoveCapture={[Function]}
          onTouchStart={[Function]}
          onTouchStartCapture={[Function]}
          role="button"
          tabIndex={0}
        >
          <Component>
            <svg
              fill="none"
              height="20px"
              stroke="#212529"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="20px"
            >
              <polygon
                points="13 19 22 12 13 5 13 19"
              />
              <polygon
                points="2 19 11 12 2 5 2 19"
              />
            </svg>
          </Component>
        </div>
      </Button>
    </div>
  </Layer>
</Controls>
`;
